packages/react-testing/src/element.ts

```diff
@@ -252,3 +261,18 @@ function equalSubset(subset: object, full: object) {
     key => key in full && full[key] === subset[key],
   );
 }
+
+function getDescendants(element: any) {
+  const descendants: Element<unknown>[] = [];
+  // eslint-disable-next-line @typescript-eslint/prefer-for-of
+  for (let i = 0; i < element.allChildren.length; i++) {
```

Comment:
I wonder if we could change our compilation settings for this package to have the foreach version of this not get transpiled, this is a node library, and node definitely supports foreach.

Reply:
I've tried targeting `ES2015` and changing that function to a more simplified: 

```
function getDescendants(element: any) {
  const descendants: Element<unknown>[] = [];
  for (const child of element.allChildren) {
    if (typeof child !== 'string') {
      descendants.push(child, ...child.elementDescendants);
    }
  }

  return descendants;
}
```

but it is unfortunately considerably slower :(

`for of` with spread operator: 

<img width="1011" alt="Screen Shot 2021-04-05 at 11 42 26 AM" src="https://user-images.githubusercontent.com/3513040/113593235-2df50500-9604-11eb-96d6-9f0243f88e5a.png">

for loop with no spread:

<img width="1011" alt="Screen Shot 2021-04-05 at 11 42 05 AM" src="https://user-images.githubusercontent.com/3513040/113593300-48c77980-9604-11eb-9b5a-9942ae531233.png">

(There's a slight variance between runs. I run these tests several times, wait for them to stabilize and take the lowest run time)